roles/download/defaults/main.yml

```diff
@@ -422,6 +426,22 @@ downloads:
     version: "{{ vault_version }}"
     groups:
       - vault
+  ingress_nginx_controller:
+    enabled: "{{ ingress_nginx_enabled }}"
+    container: true
+    repo: "{{ ingress_nginx_controller_image_repo }}"
+    tag: "{{ ingress_nginx_controller_image_tag }}"
+    sha256: "{{ ingress_nginx_controller_digest_checksum|default(None) }}"
+    groups:
+      - kube-ingress
```

Comment:
what if you dont use the kube-ingress node? Better to use kube-node?

Reply:
My original (poor) thought: ReplicaSet + nodeSelector = bad idea, can’t ensure only 1 pod per target node

First accepted implantation with your suggestion: DeamonSet for ALL node = good idea for HA

Second accepted implantation (without me): add *optional* group kube-ingress for NodeSelector on DaemonSet, if not specify then fail back to ALL node... I have not much comment for this...

But now when revamp with download role, such optional for kube-ingress make us difficult to determine which node should *exactly* pre-download the image:
* If simply say kube-ingress, image will not download if user don’t define it
* If simply say kube-node, image may download to non-kube-ingress node (which not required for) if kube-ingress is defined...

So here my answer is: making kube-ingress A MUST for deploying ingress-nginx, so no more confusion :-)